compiler/rustc_hir/src/hir.rs

```diff
@@ -2898,12 +2898,12 @@ impl ForeignItemId {
 /// passes to find the impl they want without loading the ID (which
 /// means fewer edges in the incremental compilation graph).
 #[derive(Debug, HashStable_Generic)]
-pub struct ForeignItemRef<'hir> {
+pub struct ForeignItemRef {
     pub id: ForeignItemId,
     #[stable_hasher(project(name))]
     pub ident: Ident,
     pub span: Span,
-    pub vis: Visibility<'hir>,
+    pub is_pub: bool,
```

Comment:
I don't think it's necessary to keep the `is_pub` flags in HIR.
It's used only in couple of places and the same information is available through `tcx.visibility(def_id)`.